pkg/payload/precondition/precondition.go

```diff
@@ -31,7 +31,7 @@ func (e *Error) Cause() error {
 // Precondition defines the precondition check for a payload.
 type Precondition interface {
 	// Run executes the precondition checks ands returns an error when the precondition fails.
-	Run(context.Context) error
+	Run(ctx context.Context, desiredVersion string) error
```

Comment:
Probably better to pass a concrete struct here like `ReleaseContext struct { DesiredVersion types.Version }`